src/lib.rs

```diff
@@ -672,12 +665,24 @@ impl FormatModule {
             Format::Slice(_expr, a) => self.infer_format_type(scope, a),
             Format::Bits(a) => self.infer_format_type(scope, a),
             Format::WithRelativeOffset(_expr, a) => self.infer_format_type(scope, a),
+            Format::Map(a, expr) => {
```

Comment:
Do we care at all if the expansion of the lambda is independent of the type of the value it binds? Or do we always want to make sure that type inference on the bound variable resolves even if it is not relevant to the mapped value type?

Reply:
We still want to perform type inference on the format being mapped, even if the value is ignored.